driver/src/solution_submission/mod.rs

```diff
@@ -106,7 +108,133 @@ impl<'a> StableXSolutionSubmitting for StableXSolutionSubmitter<'a> {
         claimed_objective_value: U256,
     ) -> Result<(), SolutionSubmissionError> {
         self.contract
-            .submit_solution(batch_index, orders, solution, claimed_objective_value)
-            .map_err(SolutionSubmissionError::from)
+            .submit_solution(
+                batch_index.clone(),
+                orders.clone(),
+                solution.clone(),
+                claimed_objective_value,
+            )
+            .map_err(|err| {
+                extract_transaction_hash(&err)
+                    .and_then(|hash| {
+                        let receipt = self.rpc.get_transaction_receipts(hash).ok()??;
```

Comment:
`??!?!?!`

Reply:
is there a better way? I'm getting a Result<Option<T>> and only want to continue if the inner option is one. Otherwise return None, ideally without adding another layer of nestedness.